app/components/geolocation/importer.rb

```diff
@@ -0,0 +1,60 @@
+require 'csv'
+
+module Geolocation
+  class Importer
+    BATCH_SIZE = 1000
+
+    def initialize(filepath:, batch_size: BATCH_SIZE, repo: Persistence::Repos::LocationsRepo.new(Config.rom))
+      @filepath = filepath
+      @batch_size = batch_size
+      @repo = repo
+    end
+
+    def call
+      stats, time_elapsed = measure_time do
+        CSV.foreach(@filepath, headers: true).each_slice(@batch_size).reduce(initial_stats) do |acc, rows_batch|
```

Comment:
the idea here is to read the CSV file line by line (using `CSV.foreach`), accumulate a batch of records, and bulk insert them into the DB. 
* If row has invalid data, it's not being saved in DB.
* If row has an IP-address which is already present in the DB, it's not being saved